pydicom/dataset.py

```diff
@@ -804,40 +756,57 @@ def convert_pixel_data(self):
-                handlers_tried = " ".join(
-                    [str(x) for x in pydicom.config.image_handlers])
-                logger.info("%s did not support this transfer syntax",
-                            handlers_tried)
-                self._pixel_array = None
-                self._pixel_id = None
-                if last_exception:
-                    raise last_exception
-                else:
-                    msg = ("No available image handler could "
-                           "decode this transfer syntax {}".format(
-                               self.file_meta.TransferSyntaxUID.name))
-                    raise NotImplementedError(msg)
-            # is this guaranteed to work if memory is re-used??
-            self._pixel_id = id(self.PixelData)
+        if already_have:
+            return
+
+        # Find all pixel data handlers that support the transfer syntax
+        suitable_handlers = [hh for hh in pydicom.config.image_handlers
+                             if hh and hh.supports_transfer_syntax(self)]
+
+        # No suitable handlers are available
+        if not suitable_handlers:
+            raise NotImplementedError(
+                "Unable to decode pixel data with a transfer syntax UID of "
+                "'{0}' ({1}) as there are no suitable pixel data handlers "
+                "available.".format(self.file_meta.TransferSyntaxUID,
+                                    self.file_meta.TransferSyntaxUID.name)
+            )
+
+        last_exception = None
+        for handler in suitable_handlers:
+            try:
+                # Use the handler to get a 1D numpy array of the pixel data
+                arr = handler.get_pixeldata(self)
+                self._pixel_array = reshape_pixel_array(self, arr)
+
+                # Some handler/transfer syntax combinations may need to
+                #   convert the colour space from YCbCr to RGB
+                if handler.needs_to_convert_to_RGB(self):
+                    self._pixel_array = convert_colour_space(self._pixel_array,
+                                                             'YBR_FULL',
+                                                             'RGB')
+
+                # is this guaranteed to work if memory is re-used??
+                self._pixel_id = id(self.PixelData)
+
+                return
+            except Exception as exc:
+                logger.debug(
+                    "Exception raised by pixel data handler", exc_info=exc
+                )
+                last_exception = exc
+
+        # The only way to get to this point is if we failed to get the pixel
+        #   array because all suitable handlers raised exceptions
+        self._pixel_array = None
+        self._pixel_id = None
+
+        logger.info(
+            "Unable to decode the pixel data using the following handlers: {}"
+            .format(", ".join([str(hh) for hh in suitable_handlers]))
+        )
+
```

Comment:
Perhaps here the link to 'Supported Transfer Syntaxes' table noted above could be added again - perhaps the user could try a different one not yet installed.

Reply:
Done